tests/cephadm/test_node_safe_stop.py

```diff
@@ -0,0 +1,71 @@
+from ceph.ceph import CommandFailed
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class IncorrectAlertWarningError(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verify if a host be safely stopped without reducing the availability
+     using ceph orch ok-to-stop command
+
+    Args:
+        ceph_cluster: Ceph cluster object
+        **kw :     Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful else 1 (for failure).
+    """
+
+    config = kw.get("config")
+
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+
+    installer_node = ceph_cluster.get_nodes("installer")[0].hostname
+    mds_node = rados_obj.fetch_host_node(daemon_type="mds").hostname
+    rgw_node = rados_obj.fetch_host_node(daemon_type="rgw").hostname
+
+    ok_to_stop_cmd = "cephadm shell ceph orch host ok-to-stop"
+
+    # Validate ALERT message
+    # When trying to stop the installer_node, it should raise an Alert as
+    # there is an active mgr running
+    try:
+        cmd = f"{ok_to_stop_cmd} {installer_node}"
+        _, err = cephadm.installer.exec_command(cmd, sudo=True)
+        if "ALERT: " not in err:
+            raise IncorrectAlertWarningError("No expected ALERT message shown")
+    except CommandFailed:
+        log.error("Failed to execute ok-to-stop cmd on installer_node")
+        return 1
+
+    # Validate WARNING message
+    # When trying to stop RGW node, it should raise a WARNING saying removing
+    # RGW daemon can cause clients to lose connectivity
+    try:
+        cmd = f"{ok_to_stop_cmd} {rgw_node}"
+        _, err = cephadm.installer.exec_command(cmd, sudo=True)
+        if "WARNING: " not in err:
+            raise IncorrectAlertWarningError("No expected WARNING message shown")
+    except CommandFailed:
+        log.error("Failed to execute ok-to-stop cmd on rgw node")
+        return 1
+
+    # Validate no ALERT/WARNING messages are shown when trying to remove a safe node
+    try:
+        cmd = f"{ok_to_stop_cmd} {mds_node}"
+        _, err = cephadm.installer.exec_command(cmd, sudo=True)
+        if "WARNING" in err or "ALERT" in err:
+            raise IncorrectAlertWarningError("Unexpected WARNING/ALERT message shown")
+    except CommandFailed:
+        log.error("Failed to execute ok-to-stop cmd on mds node")
+        return 1
```

Comment:
Could we optimize? For example, the difference being the type/role of the node... so we could use a `for loop statement` like
```
roles = ["installer", "rgw", "mds"]
for role in roles:
  node = ceph_cluster.get_nodes(role=role)[0]
  _shortname = node.shortname
  _, err = cephadm.installer.exec_command(cmd=f"{ok_to_stop_cmd} _shortname", sudo=True)
  ...
```